ivy/functional/backends/paddle/experimental/elementwise.py

```diff
@@ -177,7 +177,21 @@ def nan_to_num(
     neginf: Optional[Union[float, int]] = None,
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    raise IvyNotImplementedException()
+    if copy:
+        x = x.clone()
+    nan_mask = paddle.isnan(x)
+    if nan_mask.any():
+        x[nan_mask] = nan
+    if posinf is not None:
+        posinf_mask = paddle.isinf(x) & (x > 0)
+        if posinf_mask.any():
+            x[posinf_mask] = posinf
+    if neginf is not None:
+        neginf_mask = paddle.isinf(x) & (x < 0)
+        if neginf_mask.any():
+            x[neginf_mask] = neginf
+    return x
```

Comment:
here you cloned x and returned it in x which isn't considered cloning since the original variable changed, also I would suggest using an implementation similar to tensorflow backend, it's much cleaner than this one since paddle.any doesn't work with alot of datatypes